snapcraft/internal/yaml.py

```diff
@@ -369,28 +369,32 @@ def _build_env(root, arch_triplet):
         include_path_for_env = _get_include_paths(envvar, root, arch_triplet)
         if include_path_for_env:
             env.append(include_path_for_env)
+    include_path_for_env = _get_include_paths(
+        'CPATH', root, arch_triplet, prepend='', sep=':')
+    if include_path_for_env:
+        env.append(include_path_for_env)
     library_path = _get_library_paths('LDFLAGS', root, arch_triplet)
     if library_path:
         env.append(library_path)
+    library_path = _get_library_paths(
+        'LIBRARY_PATH', root, arch_triplet, prepend='', sep=':')
+    if library_path:
+        env.append(library_path)
```

Comment:
@sergiusens @elopio may I draw your attention here. qmake only seems to respect these flags instead of the gcc ones. I had to put them here because I didn't want to duplicate the `_get_*_paths()` logic in the plugin, and that logic is not public. Is this okay? Or should we make those functions part of the public API?